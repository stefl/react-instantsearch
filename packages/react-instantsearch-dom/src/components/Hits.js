import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import createClassNames from '../core/createClassNames';

const cx = createClassNames('Hits');

const Hits = ({ hits, className, hitComponent: HitComponent }) => (
  // Spread the hit on HitComponent instead of passing the full object. BC.
  // ex: <HitComponent {...hit} key={hit.objectID} />
  <div className={classNames(cx(''), className)}>
    <ul className={cx('list')}>
      {hits.map(hit => (
        <li className={cx('item')} key={hit.objectID}>
          <HitComponent hit={hit} />
        </li>
      ))}
    </ul>
  </div>
);

Hits.propTypes = {
  hits: PropTypes.arrayOf(PropTypes.object).isRequired,
  className: PropTypes.string,
  hitComponent: PropTypes.func,
};

Hits.defaultProps = {
  className: '',
  hitComponent: props => (
    <div
      style={{
        borderBottom: '1px solid #bbb',
        paddingBottom: '5px',
        marginBottom: '5px',
        wordBreak: 'break-all',
      }}
    >
      {JSON.stringify(props).slice(0, 100)}...
    </div>
  ),
};

export default Hits;
