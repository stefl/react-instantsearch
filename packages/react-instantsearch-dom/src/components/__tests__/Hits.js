import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import Hits from '../Hits';

describe('Hits', () => {
  const Hit = ({ hit }) => <div id={hit.objectID} />;

  Hit.propTypes = {
    hit: PropTypes.object,
  };

  it('accepts a hitComponent prop', () => {
    const hits = [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }];

    const tree = renderer.create(<Hits hitComponent={Hit} hits={hits} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('accepts a custom className', () => {
    const hits = [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }];

    const tree = renderer.create(
      <Hits className="MyCustomHits" hitComponent={Hit} hits={hits} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
