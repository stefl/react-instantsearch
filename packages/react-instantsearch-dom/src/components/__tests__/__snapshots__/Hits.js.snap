// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits accepts a custom className 1`] = `
<div
  className="ais-Hits MyCustomHits"
>
  <ul
    className="ais-Hits-list"
  >
    <li
      className="ais-Hits-item"
    >
      <div
        id={0}
      />
    </li>
    <li
      className="ais-Hits-item"
    >
      <div
        id={1}
      />
    </li>
    <li
      className="ais-Hits-item"
    >
      <div
        id={2}
      />
    </li>
  </ul>
</div>
`;

exports[`Hits accepts a hitComponent prop 1`] = `
<div
  className="ais-Hits"
>
  <ul
    className="ais-Hits-list"
  >
    <li
      className="ais-Hits-item"
    >
      <div
        id={0}
      />
    </li>
    <li
      className="ais-Hits-item"
    >
      <div
        id={1}
      />
    </li>
    <li
      className="ais-Hits-item"
    >
      <div
        id={2}
      />
    </li>
  </ul>
</div>
`;
